### PR TITLE
PIN-1341 authorization-process POST /clients/{clientId}/relationships/{relationshipId}

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -6512,6 +6512,108 @@ paths:
...
           type: string
           format: uuid
+    post:
+      tags:
+        - clients
+      summary: Binds an Operator relationship to a Client
+      description: Binds a security operator belonging to a consumer to a Client
+      operationId: clientOperatorRelationshipBinding
+      responses:
+        '201':
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
        '200':
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala [hunk 2]
```diff
@@ -84,4 +84,24 @@ final case class ClientsApiServiceImpl(authorizationProcessService: Authorizatio
...
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerCreatedResource: ToEntityMarshaller[CreatedResource]
+  ): Route = {
+
+    val result: Future[CreatedResource] = for {
+      clientUuid       <- clientId.toFutureUUID
+      relationshipUuid <- relationshipId.toFutureUUID
+      result           <- authorizationProcessService.clientOperatorRelationshipBinding(clientUuid, relationshipUuid)
+    } yield (CreatedResource(result.id))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala]: which id is returned here, the client or the relationship?

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala [hunk 3]
```diff
@@ -84,4 +84,24 @@ final case class ClientsApiServiceImpl(authorizationProcessService: Authorizatio
...
+
+    onComplete(result) {
+      handleError(s"Error binding operator relationship $relationshipId of client $clientId") orElse {
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala]:
```suggestion
      handleError(s"Error binding operator relationship $relationshipId to client $clientId") orElse {
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/AuthorizationProcessServiceImpl.scala [hunk 4]
```diff
@@ -68,4 +69,18 @@ class AuthorizationProcessServiceImpl(authorizationProcessUrl: String, blockingE
...
+  ): Future[Client] =
+    withHeaders[Client] { (bearerToken, correlationId, ip) =>
+      val request: ApiRequest[Client] =
+        api.clientOperatorRelationshipBinding(
+          clientId = clientId,
+          relationshipId = relationshipId,
+          xCorrelationId = correlationId,
+          xForwardedFor = ip
+        )(BearerToken(bearerToken))
+      invoker.invoke(request, s"Binding operator relationship $relationshipId of client ${clientId.toString}")
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/AuthorizationProcessServiceImpl.scala]:
```suggestion
      invoker.invoke(request, s"Binding operator relationship $relationshipId to client ${clientId.toString}")
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala (hunk 2 above)
Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala]: The client

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala]: Is specializing the return type a good idea here maybe?

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala (hunk 2 above)
Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala]: Yes, sure.

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**